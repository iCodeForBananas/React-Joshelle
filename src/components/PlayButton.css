.btn3d {
  position: relative;
  border: 0;
  transition: all 40ms linear;
  margin-top: 25%;
  font-size: 50px;
  margin-bottom: 10px;
  margin-left: 30%;
  margin-right: 2px;
}

@media only screen and (max-width: 440px) {
  .btn3d {
    position: relative;
    font-size: 40px;
    margin: 30px auto 0;
    display: block;
  }
}

.btn3d:active:focus,
.btn3d:focus:hover,
.btn3d:focus {
  outline: medium none;
}

.btn3d:active,
.btn3d.active {
  top: 2px;
}

.btn3d.btn-white {
  color: #666666;
  box-shadow: 0 0 0 1px #ebebeb inset, 0 0 0 2px rgba(255, 255, 255, 0.1) inset, 0 8px 0 0 #f5f5f5,
    0 8px 8px 1px rgba(0, 0, 0, 0.2);
  background-color: #fff;
}

.btn3d.btn-white:active,
.btn3d.btn-white.active {
  color: #666666;
  box-shadow: 0 0 0 1px #ebebeb inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.1);
  background-color: #fff;
}

.btn3d.btn-default {
  color: #666666;
  box-shadow: 0 0 0 1px #ebebeb inset, 0 0 0 2px rgba(255, 255, 255, 0.1) inset, 0 8px 0 0 #bebebe,
    0 8px 8px 1px rgba(0, 0, 0, 0.2);
  background-color: #f9f9f9;
}

.btn3d.btn-default:active,
.btn3d.btn-default.active {
  color: #666666;
  box-shadow: 0 0 0 1px #ebebeb inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.1);
  background-color: #f9f9f9;
}

.btn3d.btn-primary {
  box-shadow: 0 0 0 1px #417fbd inset, 0 0 0 2px rgba(255, 255, 255, 0.15) inset, 0 8px 0 0 #4d5bbe,
    0 8px 8px 1px rgba(0, 0, 0, 0.5);
  background-color: #4274d7;
}

.btn3d.btn-primary:active,
.btn3d.btn-primary.active {
  box-shadow: 0 0 0 1px #417fbd inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.3);
  background-color: #4274d7;
}

.btn3d.btn-success {
  box-shadow: 0 0 0 1px #31c300 inset, 0 0 0 2px rgba(255, 255, 255, 0.15) inset, 0 8px 0 0 #5eb924,
    0 8px 8px 1px rgba(0, 0, 0, 0.5);
  background-color: #78d739;
}

.btn3d.btn-success:active,
.btn3d.btn-success.active {
  box-shadow: 0 0 0 1px #30cd00 inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.3);
  background-color: #78d739;
}

.btn3d.btn-info {
  box-shadow: 0 0 0 1px #00a5c3 inset, 0 0 0 2px rgba(255, 255, 255, 0.15) inset, 0 8px 0 0 #348fd2,
    0 8px 8px 1px rgba(0, 0, 0, 0.5);
  background-color: #39b3d7;
}

.btn3d.btn-info:active,
.btn3d.btn-info.active {
  box-shadow: 0 0 0 1px #00a5c3 inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.3);
  background-color: #39b3d7;
}

.btn3d.btn-warning {
  box-shadow: 0 0 0 1px #d79a47 inset, 0 0 0 2px rgba(255, 255, 255, 0.15) inset, 0 8px 0 0 #d79a34,
    0 8px 8px 1px rgba(0, 0, 0, 0.5);
  background-color: #feaf20;
}

.btn3d.btn-warning:active,
.btn3d.btn-warning.active {
  box-shadow: 0 0 0 1px #d79a47 inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.3);
  background-color: #feaf20;
}

.btn3d.btn-danger {
  box-shadow: 0 0 0 1px #b93802 inset, 0 0 0 2px rgba(255, 255, 255, 0.15) inset, 0 8px 0 0 #aa0000,
    0 8px 8px 1px rgba(0, 0, 0, 0.5);
  background-color: #d73814;
}

.btn3d.btn-danger:active,
.btn3d.btn-danger.active {
  box-shadow: 0 0 0 1px #b93802 inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.3);
  background-color: #d73814;
}

.btn3d.btn-magick {
  color: #fff;
  box-shadow: 0 0 0 1px #9a00cd inset, 0 0 0 2px rgba(255, 255, 255, 0.15) inset, 0 8px 0 0 #9823d5,
    0 8px 8px 1px rgba(0, 0, 0, 0.5);
  background-color: #bb39d7;
}

.btn3d.btn-magick:active,
.btn3d.btn-magick.active {
  box-shadow: 0 0 0 1px #9a00cd inset, 0 0 0 1px rgba(255, 255, 255, 0.15) inset, 0 1px 3px 1px rgba(0, 0, 0, 0.3);
  background-color: #bb39d7;
}
