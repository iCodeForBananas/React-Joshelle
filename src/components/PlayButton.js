import React from "react";
import "./PlayButton.css";

type Props = {
  onClick: () => void
};

const PlayButton = function({ onClick }: Props) {
  return (
    <button type="button" className="btn btn-danger btn-lg btn3d" onClick={onClick}>
      Start Joshelleing!
    </button>
  );
};

export default PlayButton;
