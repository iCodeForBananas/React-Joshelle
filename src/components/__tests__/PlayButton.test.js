import React from "react";
import PlayButton from "../PlayButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<PlayButton onClick={() => null} />).toJSON();
  expect(tree).toMatchSnapshot();
});
