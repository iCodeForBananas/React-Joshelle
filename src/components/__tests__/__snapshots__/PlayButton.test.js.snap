// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-danger btn-lg btn3d"
  onClick={[Function]}
  type="button"
>
  Start Joshelleing!
</button>
`;
